modules/hal_infineon/mtb-hal-cat1/CMakeLists.txt

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set(hal_dir               ${ZEPHYR_HAL_INFINEON_MODULE_DIR}/mtb-hal-cat1)
+set(hal_psoc6_dir         ${hal_dir}/COMPONENT_PSOC6HAL)
+set(hal_psoc6_cat1a_dir   ${hal_dir}/COMPONENT_PSOC6HAL/COMPONENT_CAT1A)
```

Comment:
to preserve bisectability, HAL module needs to come before drivers or any other code.

Reply:
rebased